src/external-repository/scripts/setup-user.sh

```diff
@@ -78,9 +85,6 @@ configure_git_for_user() {
         # before this script is called and it will still work. This allows for other techniques to be
         # used to communicate the desire to checkout a branch
         checkout_branch ${WC_PATH}
-    else
-        echo "Configuring Git Credentials to use a secret"
-        cat "./usersecret-git.config" >> "${GIT_PATH}/config"
     fi
 
     # Setup Git Telemetry .. note that the checks for the credentials
```

Comment:
Note that this code, and possibly some other code in this script is operating under the assumption the script will have returned if the credential helper has already been setup. In other words, there is an expectation in this script that it only runs one time. It is possible you can use the new `credential_configured` variable to restore some of these expectations but you need to review the entire flow of the script and what it is doing more carefully

Reply:
Right the branch feature is only there when a new git repos is created, which should be never.

There is no way to pass in data when creating a Codespace so passing in a branch name is not really possible. The feature you have been editing works by having a branch with a special prefix in the Codespace repository. If you then create a Codespace on that branch at runtime it detects the branch name and parses off the prefix to get the branch name you want to open from the external repository and checks out that branch after cloning the repo. It is just a hack.